### ISSUE TITLE
Phase Shift

### ISSUE THREAD

**Issue by crossan007** [krzychb/dac-cosine#3]
THANKS so much for this API, It works very nicely.

Is it possible to implement a phase shift on the generated wave?

**Comment by krzychb** [comment]
@crossan007 I am happy you found it useful!
Only 180° phase shift is possible.
If you are using ESP-IDF then the API is already there https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/dac.html?highlight=cosine#_CPPv423dac_cw_generator_enablev

**Comment by crossan007** [comment]
Makes sense - thank for the link.  

I was able to generate a 60khz sine wave using your API wrapper here fairly easily, but with the limited phase-shift options available in the ESP hardware, I'm shifting my focus a bit to DMA as a sample source for the DAC.  I think I should be able to phase-shift the sample array this way?    

I've found a bunch of "sample code" for writing "buffers of samples" to the DAC using the ESP I2S library (which reportedly uses DMA), but I have been unable to get any of these libraries to work.  (I realize this is outside the scope of this repository; just sharing some context for my project) 

For context; this is what I'm trying to accomplish: https://github.com/crossan007/ATTiny85WWVBReceiver/
